lib/secretsManager.js

```diff
@@ -1,77 +1,169 @@
+        await SecureStore.deleteItemAsync(key)
+      }
+      return
+    }
+
+    await AsyncStorage.multiRemove([
+      encryptionPassphraseStorageKey,
+      secretStorageKey,
+    ])
+  }
 
-/**
- * Add or update a secret to the list.
- * If not provided, an _id key will be generated
- * @async
- * @param {Object} secret - secret to upsert
- * @returns {Object} the added/updated secret
- */
-export async function upsert(secret) {
-  const secretId = secret._id || DUMMY_UUID_GEN++
-  const secrets = await find()
-  let upserted = secrets.find(({ _id }) => _id === secretId)
-
-  if (upserted) {
-    Object.assign(upserted, secret)
-  } else {
-    upserted = { _id: secretId, ...secret }
-    secrets.push(upserted)
+  async encrypt(secrets) {
+    const secretsBytes = aesjs.utils.utf8.toBytes(JSON.stringify(secrets))
+    const aesCtr = new aesjs.ModeOfOperation.ctr(
+      Uint8Array.from(JSON.parse(this.encryptionPassphrase)),
+      new aesjs.Counter(5)
+    )
+    const encryptedBytes = aesCtr.encrypt(secretsBytes)
+    return aesjs.utils.hex.fromBytes(encryptedBytes)
   }
 
-  try {
-    DUMMY_STORE = secrets
-  } catch (err) {
-    console.error('Failed to add new secret to data store', err)
+  async decrypt(encryptedSecrets) {
+    const encryptedBytes = aesjs.utils.hex.toBytes(encryptedSecrets)
+    const aesCtr = new aesjs.ModeOfOperation.ctr(
+      Uint8Array.from(JSON.parse(this.encryptionPassphrase)),
+      new aesjs.Counter(5)
+    )
+    const decryptedBytes = aesCtr.decrypt(encryptedBytes)
+    return JSON.parse(aesjs.utils.utf8.fromBytes(decryptedBytes))
   }
 
-  return upserted
-}
+  async get() {
+    const encrypted = await AsyncStorage.getItem(secretStorageKey)
+    return encrypted ? this.decrypt(encrypted) : []
+  }
+
+  async set(secrets) {
+    return AsyncStorage.setItem(secretStorageKey, await this.encrypt(secrets))
+  }
+
+  /**
+   * Find all secrets, applying a selector if provided.
+   * When provided, all where clause must be fulfilled (logical AND).
+   * By default, returns all secrets
+   * @async
+   * @param {Object} where - hash of where clauses returned secrets must fulfill with
+   * @returns {Array<Object>} list (may be empty) of matching secrets
+   */
+  async find(where = {}) {
+    let secrets = []
 
-/**
- * Remove a secret from the list. Does not fail if it can not be found.
- * @async
- * @param {String} secretId - removed secret id
- * @returns {Object} the removed secret (may be null)
- */
-export async function remove(secretId) {
-  const secrets = await find()
-  const idx = secrets.findIndex(({ _id }) => _id === secretId)
-  let removed = null
-
-  if (idx !== -1) {
-    removed = secrets.splice(idx, 1)[0]
+    try {
+      const props = Object.keys(where)
+      secrets = (await this.get()).filter(secret =>
+        props.every(prop => secret[prop] === where[prop])
+      )
+    } catch (err) {
+      console.error('Failed to read secrets from data store', err)
```

Comment:
not for now, but we may want to improve error handling at some point. please create an issue for this

Reply:
https://github.com/nearform/optic-expo/issues/36